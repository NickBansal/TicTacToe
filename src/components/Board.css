.GameBoard {
    width: 400px;
    height: 400px;
    margin: 0 auto;
    background: #34495e;
    color: #fff;
    border: 6px solid #2c3e50;
    border-radius: 10px;
    display: grid;
    grid-template: repeat(3, 1fr) / repeat(3, 1fr);
}

.Box {
    border: 6px solid #2c3e50;
    border-radius: 2px;
    font-family: Helvetica;
    font-weight: bold;
    font-size: 6em;
    display: flex;
    justify-content: center;
    align-items: center;
}

.Box:hover {
    cursor: pointer;
    background: #44505c;
}