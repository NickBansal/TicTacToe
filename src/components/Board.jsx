import React from 'react';
import './Board.css'

const Board = ({ gameSquares, handleClick }) => {
    return (
        <div className="GameBoard">
            {gameSquares.map((element, index) => {
                return <div 
                key={index}
                onClick = {() => handleClick(index)}
                className="Box">{element}</div>
            })}
        </div>
    )
}


export default Board;