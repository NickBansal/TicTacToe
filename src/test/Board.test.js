import './setupTest';
import React from 'react';
import Board from '../components/Board';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe.only('<Board />', () => {

    const minProps = {
        gameSquares: Array(9).fill(null), 
        handleClick: () => {}
    }

    let wrapper
    beforeEach(() => {
        wrapper = shallow(<Board {...minProps}/>)
    })
    it('Component matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    it('Checks that the div component with className GameBoard exists', () => {
        expect(wrapper.find('div.GameBoard').exists()).toBe(true)
    })
    it('Counts the amount of squares on the gameboard', () => {
        expect(wrapper.find('div.Box').length).toBe(9)
    })
})