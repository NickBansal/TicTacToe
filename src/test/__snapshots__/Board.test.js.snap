// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> Component matches the snapshot 1`] = `
<div
  className="GameBoard"
>
  <div
    className="Box"
    key="0"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="1"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="2"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="3"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="4"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="5"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="6"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="7"
    onClick={[Function]}
  />
  <div
    className="Box"
    key="8"
    onClick={[Function]}
  />
</div>
`;
